pkg/loop/prom.go

```diff
@@ -0,0 +1,109 @@
+package loop
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"golang.org/x/net/context"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+)
+
+type promServer struct {
+	port        int
+	srvrDone    chan struct{} // closed when the http server is done
+	srvr        *http.Server
+	tcpListener *net.TCPListener
+	lggr        logger.Logger
+
+	handler http.Handler
+}
+
+type promServerOpt func(*promServer)
+
+func withHandler(h http.Handler) promServerOpt {
+	return func(s *promServer) {
+		s.handler = h
+	}
+}
+
+// TODO there is one core/web test using this....
```

Comment:
I wanted to avoid exporting this, since the common helper code can handle it internally, but there is still one test in core using it. We may be able to do the same validation indirectly though :thinking:

Reply:
The core test is exercising the loop registration and service discovery.

I think we need that test, because is it is only test for the metric plumbing between the two sides.

Part of the contract btw the core node and a LOOPP is that a LOOPP ought to expose a prom server. We could promote that to a real interface somehow and then mock on the core side test for the registry.

or this repo could expose a mock/test implementation that runs a prom server that could be used for that test. 

or we could leave this public and keep the test as is.

i think 1 is the most principled thing to do, but doesn't fit in this PR, so maybe 2 is the best immediate option?

Reply:
Exported again.